crates/hc/src/save.rs

```diff
@@ -96,3 +97,65 @@ pub fn list(hc_dir: PathBuf, verbose: usize) -> anyhow::Result<()> {
     msg!("{}", out);
     Ok(())
 }
+
+lazy_static::lazy_static! {
+    static ref FILE_LOCKS: tokio::sync::Mutex<Vec<usize>> = tokio::sync::Mutex::new(Vec::new());
+}
+
+/// Lock this setup as running live and advertise the port
+pub async fn save_port(mut hc_dir: PathBuf, path: &Path, port: u16) -> anyhow::Result<()> {
+    use std::io::Write;
+    std::fs::create_dir_all(&hc_dir)?;
+    let paths = load(hc_dir.clone())?;
+    let index = match paths.into_iter().enumerate().find(|p| p.1 == path) {
+        Some((i, _)) => i,
+        None => return Ok(()),
+    };
+    hc_dir.push(format!(".hc_live_{}", index));
+    match std::fs::OpenOptions::new()
+        .write(true)
+        .create_new(true)
+        .open(hc_dir)
+    {
+        Ok(mut file) => {
+            writeln!(file, "{}", port)?;
```

Comment:
I don't know what are policies and context around it, but writing directly to a file leads to potential corrupted data in case of crashes and errors. Generally files should be written to using a "write to a tempfile + rename" pattern. [Example](https://github.com/crev-dev/cargo-crev/blob/bd7b8407a2c78c8a263e3b4f827ade4a3c00b4da/crev-common/src/lib.rs#L330) from one of my apps.

Reply:
The context is for creating temporary holochain setups to run an app in. I see where you're coming from but it's really not a big deal if the file was to be corrupted. You'd just run `hc clean` and start again. I don't have a strong opinion though.